olp-cpp-sdk-authentication/src/Crypto.cpp

```diff
@@ -61,19 +63,33 @@ static uint32_t SHA256_K[SHA256_CONSTANTS_LENGTH] = {
 #define HMAC_OPAD_BYTE 0x5c
 #define HMAC_B 64
 
-namespace olp {
-namespace authentication {
-std::vector<uint8_t> Crypto::hmac_sha256(const std::string& key,
-                                         const std::string& message) {
+class CryptoImpl {
+ public:
+  static Crypto::Sha256Digest HmacSha256(const std::string& key,
+                                         const std::string& message);
+
+  static Crypto::Sha256Digest Sha256(const std::vector<unsigned char>& src);
+
+ private:
+  static std::vector<uint32_t> Sha256Init();
+  static void Sha256Transform(const std::vector<unsigned char>& currentChunk,
+                              unsigned long startIndex,
+                              std::vector<uint32_t>& hashValue);
+  static std::vector<unsigned char> ToUnsignedCharVector(
+      const std::string& src);
```

Comment:
Highly unusual. Why not make them anonymous functions and drop the CryptoImpl class?